### PR TITLE
fix: trap on overflow in `i32::checked_div`

### LINKED ISSUE #1162: Incorrect edge case handling in intrinsic `i32::checked_div`
`proc i32::checked_div` is supposed to trap on overflow, however it does not trap on inputs `i32::MIN, -1`.

### Reproduction

Requires #1160 

```sh
cargo test -p midenc-integration-tests i32_checked_div_min_by_neg1 -- --nocapture
```

### PR DESCRIPTION
**Pull request by mooori**
Closes #1162 

Stacked on top of #1160 to have access to intrinsic testing.

---

The intrinsic `i32::checked_div` should trap on `i32::MIN/-1` because the result overflows `i32`. It did not trap because it didn't account for this edge case.

### CODE REVIEW

File: tests/integration/src/end_to_end/intrinsics/i32_arithmetic.rs [hunk 1]
```diff
@@ -550,43 +550,3 @@ fn i32_checked_shr() {
         },
     );
 }
-
-// TODO(#1162): remove once #1162 is fixed and handle MIN/-1 in `i32_checked_div` above
-/// This test reproduces #1162
-#[test]
-fn i32_checked_div_min_by_neg1_i1162_reproducer() {
```
Comment (mooori) [inline tests/integration/src/end_to_end/intrinsics/i32_arithmetic.rs:557]: Can be removed as the test `i32_checked_div` now asserts that `(i32::MIN, -1)` causes a `FailedAssertionOverflow` trap.

Comment (greenhat) [reply tests/integration/src/end_to_end/intrinsics/i32_arithmetic.rs:557]: It is one weighted branch in a randomized `prop_oneof!`, so a proptest run can pass without executing the exact `i32::MIN / -1` case.

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking good! Please see my comment.

File: tests/integration/src/end_to_end/intrinsics/i32_arithmetic.rs (hunk 1 above)
Comment (mooori) [reply tests/integration/src/end_to_end/intrinsics/i32_arithmetic.rs:557]: I assumed weights are chosen such that the probability of each case being executed is "high enough". Though your comment made me take another look as this is relevant for all tests that use `NumericStrategy`. I've opened #1169 to address this. WDYT?

**Review by bitwalker — APPROVED** [review]

**Review by greenhat — APPROVED** [review]
